sfs/time/drivingfunction.py

```diff
@@ -251,7 +252,191 @@ def apply_delays(signal, delays):
     delays_samples = np.rint(samplerate * delays).astype(int)
     offset_samples = delays_samples.min()
     delays_samples -= offset_samples
-    out = np.zeros((delays_samples.max() + len(data), len(delays_samples)))
-    for column, row in enumerate(delays_samples):
-        out[row:row + len(data), column] = data
-    return util.DelayedSignal(out, samplerate, offset_samples / samplerate)
+    out = np.zeros((delays_samples.max() + len(signal), len(delays_samples)))
+    for channel, cdelay in enumerate(delays_samples):
+        out[cdelay:cdelay + len(signal), channel] = signal
+    return out, offset_samples / fs
+
+    
+def nfchoa_25d_plane(x0, r0, npw, max_order=None, c=None, fs=44100, normalize=True):
+    """Vritual plane wave by 2.5-dimensional NFC-HOA.
+
+    Parameters
+    ----------
+    x0 : (N, 3) array_like
+        Sequence of secondary source positions.
+    r0 : float
+        Radius of the circular secondary source distribution
+    npw : (3,) array_like
+        Unit vector (propagation direction) of plane wave.
+    max_order : int
+        Ambisonics order
+    c : float, optional
+        Speed of sound
+
+    Returns
+    -------
+    delay : float
+        Overall delay (common to all secondary source)
+    weight : float
+        Overall weight (common to all secondary sources)
+    sos : dictionary
+        Second-order section filters 
+    phaseshift : float
+        Phase shift
+
+    """
```

Comment:
Add a reference to the help message. In sfs-matlab we used eq. (10) from Spors, Kuscher, Ahrens (2011).